### DISCUSSION TITLE
Does the oqs-provider have plans to support UOV and SPHINCS+ algorithms?

### DISCUSSION THREAD

**Discussion by liujinhua2495170** [open-quantum-safe/oqs-provider#605] [General]
Hello,

Quick question, Does the oqs-provider have plans to support UOV and SPHINCS+ algorithms?

Regards
jh

**Comment by baentsch** [comment]
Hi - Sphincs+ is supported but I don't know what UOV is: Can you provide a pointer?

**Comment by liujinhua2495170** [comment]
hi, I checked the latest oqs-provider, and it supports SPHINCS-SHA2 and SPHINCS-SHAKE, but not SPHINCS+. 
UOV： https://www.uovsig.org/

**Reply by baentsch** [reply]
> it supports SPHINCS-SHA2 and SPHINCS-SHAKE, but not SPHINCS+.

For me, Sphincs+ is the summary term for all Sphincs variants, so I'm not sure what (parameter set) exactly you're referring to. Or do you mean the standardized SLH-DSA? For that, see https://github.com/open-quantum-safe/liboqs/issues/1894

Thanks for the UOV pointer. Looking at the GH repo at https://github.com/pqov/pqov the last commit was more than a year ago, so doesn't look like under active development any more. Maybe @mkannwischer can provide guidance as to its status(?)

**Reply by liujinhua2495170** [reply]
> > it supports SPHINCS-SHA2 and SPHINCS-SHAKE, but not SPHINCS+.
> 
> For me, Sphincs+ is the summary term for all Sphincs variants, so I'm not sure what (parameter set) exactly you're referring to. Or do you mean the standardized SLH-DSA? For that, see [open-quantum-safe/liboqs#1894](https://github.com/open-quantum-safe/liboqs/issues/1894)
> 
> Thanks for the UOV pointer. Looking at the GH repo at https://github.com/pqov/pqov the last commit was more than a year ago, so doesn't look like under active development any more. Maybe @mkannwischer can provide guidance as to its status(?)

Thanks for that explanation. I appreciate your time.

**Comment by mkannwischer** [comment]
On Mon, Dec 30, 2024 at 14:38 Michael Baentsch ***@***.***>
wrote:

> Thanks for the UOV pointer. Looking at the GH repo at
> https://github.com/pqov/pqov the last commit was more than a year ago, so
> doesn't look like under active development any more. Maybe @mkannwischer
> <https://github.com/mkannwischer> can provide guidance as to its status(?)
>
The NIST deadline for the second round is in mid-January and we will make a
few small updates to that implementation by then. Integration into oqs
after that would be great. Happy to help with that.   I will ping you once
the changes are done.
